### PR TITLE
refactor(useCalendar): include calendar identifier when computing heading value

### PR DESCRIPTION
**Pull request by epr3**
Hello,

This PR extends the `useDateFormatter` function to also include the `calendar` identifier in its options, so you get the month names translated into your required `locale` and also updated the utility function to be able to accept `options`.

It also updates the `useCalendar` `headingValue` to use the `calendar` identifier when computing the `fullMonth`, `fullYear` and `fullMonthAndYear`.

Plus, the `props.placeholder` function has been updated to use `isEqualMonth` so any calendar system change would update the `grid`.

cc @sadeghbarati for pointing out the bug and providing me with reproductions.

### CODE REVIEW

File: packages/radix-vue/src/Calendar/story/CalendarSelect.story.vue [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  calendar.value = pref.value!.ordering.split(' ')[0]
+}
+const value = computed(() => toCalendar(today(getLocalTimeZone()), createCalendar(calendar.value)))
+</script>
+
+<template>
+  <Story title="Calendar/Locale and Calendar Select" :layout="{ type: 'single' }">
+    <Variant title="default">
+      <div class="flex flex-col gap-4">
+        <SelectRoot :model-value="locale" @update:model-value="updateLocale">
```
Comment (sadeghbarati) [inline packages/radix-vue/src/Calendar/story/CalendarSelect.story.vue:57]:
```suggestion
        <SelectRoot v-model="locale" @update:model-value="updateLocale">
```

**Comment by sadeghbarati** [comment]
LPTM 😄💯

**Review by sadeghbarati — APPROVED** [review]
LPTM 😄💯

**Comment by zernonia** [comment]
Nice addition @epr3 ! LGTM!

**Pull request merged**